From 47a8bfcf9129615f72a827471f6ceb9d4a8774eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Wed, 1 Oct 2025 17:12:25 +0200
Subject: [PATCH 303/553] Use size of largest representation as NSImage size in
 imageFromQIcon

This matches the behavior of loading an icns file from disk.

Pick-to: 6.8 6.5
Change-Id: I08f6ba5f4a9e7ad447a9ccdaecd2ea8216276640
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 343cc77d003696ea02629ab3072b1e2918aef8da)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/gui/painting/qcoregraphics.mm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gui/painting/qcoregraphics.mm b/src/gui/painting/qcoregraphics.mm
index 96bbcd1bbd0..0917f2484db 100644
--- a/src/gui/painting/qcoregraphics.mm
+++ b/src/gui/painting/qcoregraphics.mm
@@ -153,6 +153,9 @@ QT_END_NAMESPACE
         auto *imageRep = [[NSBitmapImageRep alloc] initWithCGImage:cgImage];
         imageRep.size = image.deviceIndependentSize().toCGSize();
         [nsImage addRepresentation:[imageRep autorelease]];
+        // Match behavior of loading icns files, where the NSImage size
+        // reflects the largest representation.
+        nsImage.size = imageRep.size;
     }
 
     if (!nsImage.representations.count)
-- 
2.51.2

